### PR TITLE
Bugfix rem.oracle last_update check

### PR DESCRIPTION
**Pull request by Alladin9393**
## Change Description
* Rename namespace `eosio` -> `remoracle`
* Add the possibility to submit differently amount of pairs
* Add check last update to the `get_relevant_prices()` method

### CODE REVIEW

File: contracts/contracts/rem.oracle/include/rem.oracle/rem.oracle.hpp [hunk 1]
```diff
@@ -8,11 +8,26 @@
...
+   using eosio::singleton;
+   using eosio::seconds;
+   using eosio::name;
+   using eosio::time_point;
+   using eosio::multi_index;
    using std::string;
    using std::vector;
 
+   // A window in which producer can submit a new rate
+   static constexpr uint32_t setprice_window = 3600;
```
Comment (roman-tik) [inline contracts/contracts/rem.oracle/include/rem.oracle/rem.oracle.hpp]:
```suggestion
   static constexpr uint32_t setprice_window_seconds = 3600;
```

File: contracts/contracts/rem.oracle/src/rem.oracle.cpp [hunk 2]
```diff
@@ -47,10 +47,10 @@ namespace eosio {
       if (is_active_producer) {
          std::map<name, vector<double>> pairs_points = get_relevant_prices();
          auto majority_amount = get_majority_amount();
-         uint32_t amount_prod_points = pairs_points.begin()->second.size();
 
-         if (amount_prod_points > majority_amount) {
-            for (const auto &points: pairs_points) {
+         for (const auto &points: pairs_points) {
+            uint32_t amount_prod_points = pairs_points.at(points.first).size();
```
Comment (roman-tik) [inline contracts/contracts/rem.oracle/src/rem.oracle.cpp]: would `points.second.size()` work here?

File: contracts/contracts/rem.oracle/src/rem.oracle.cpp [hunk 3]
```diff
@@ -84,17 +84,16 @@ namespace eosio {
    std::map<name, vector<double>> oracle::get_relevant_prices() const {
       vector <name> _producers = eosio::get_active_producers();
       std::map<name, vector<double>> prices_data;
+      auto ct = current_time_point();
 
       for (const auto &producer: _producers) {
          auto it = pricedata_tbl.find(producer.value);
          if (it != pricedata_tbl.end()) {
```
Comment (roman-tik) [inline contracts/contracts/rem.oracle/src/rem.oracle.cpp]:
```suggestion
         if (it != pricedata_tbl.end() && ((ct - it->last_update) < seconds(setprice_window * 2))) {
```
`setprice_window * 2` - to accomodate a valid case when (e.q. `last_update` = 13:01 and `ct` = 14:55)

File: contracts/contracts/rem.oracle/src/rem.oracle.cpp [hunk 4]
```diff
@@ -84,17 +84,16 @@ namespace eosio {
...
 
             for (const auto &pair: pairstable_data.pairs) {
                // if a new pair is added, but the producer doesn't add the rate for new pair, its data is skipped
-               if (it->pairs_data.count(pair) == 0) {
-                  break;
-               }
-               prices_data[pair].push_back(it->pairs_data.at(pair));
+               if (ct - it->last_update <= seconds(setprice_window) || it->pairs_data.count(pair) != 0)
```
Comment (roman-tik) [inline contracts/contracts/rem.oracle/src/rem.oracle.cpp]:
```suggestion
               if (it->pairs_data.count(pair) != 0)
```

**Review by roman-tik — CHANGES_REQUESTED** [review]

File: contracts/contracts/rem.oracle/include/rem.oracle/rem.oracle.hpp (hunk 1 above)
Comment (Alladin9393) [reply contracts/contracts/rem.oracle/include/rem.oracle/rem.oracle.hpp]: @roman-tik, done.

File: contracts/contracts/rem.oracle/src/rem.oracle.cpp (hunk 2 above)
Comment (Alladin9393) [reply contracts/contracts/rem.oracle/src/rem.oracle.cpp]: @roman-tik, yes already fixed it.

File: contracts/contracts/rem.oracle/src/rem.oracle.cpp (hunk 3 above)
Comment (Alladin9393) [reply contracts/contracts/rem.oracle/src/rem.oracle.cpp]: @roman-tik, done.

File: contracts/contracts/rem.oracle/src/rem.oracle.cpp (hunk 4 above)
Comment (Alladin9393) [reply contracts/contracts/rem.oracle/src/rem.oracle.cpp]: @roman-tik, done.

**Pull request merged**